pkg/storage/bolt.go

```diff
@@ -25,17 +29,21 @@ func (b *BoltDB) Type() Storage {
 	return Bolt
 }
 
-// NewBoltDB instantiates a file-based storage instance for Bolt https://github.com/boltdb/bolt
-func NewBoltDB() (*BoltDB, error) {
-	return NewBoltDBWithFile(DBFile)
-}
-
-func NewBoltDBWithFile(filePath string) (*BoltDB, error) {
+// Init instantiates a file-based storage instance for Bolt https://github.com/boltdb/bolt
+func (b *BoltDB) Init(param interface{}) error {
```

Comment:
is `param` supposed to be generic for all storage methods?

if so...I think this needs to be variadic or use an options pattern

if not...let's make this `filePath string`

Reply:
yes, `param`  supposed to be generic for all storage methods.  the problem is we don't know what param should be. For a kv-store, it can be a file path, but for blockchain or something else, it can be a struct.

Reply:
and yes, it can be variadic or use an options pattern. but in `Init()`, we always should input a value to function, a implementation specify struct (or type), or just a `nil`. 
i have thinking through this kind of situation:
```go
func (b *BoltDB) Init(params ...interface{}) error
func (b *BoltDB) Init(params ...string) error
```
but i think `func (b *BoltDB) Init(param interface{}) error` was the best solution.
if you have any idea, please tell me. we can talk about it deeply.

Reply:
> if so...I think this needs to be variadic or use an options pattern
> 
by the way, i really should change `param` to `options`.